### PR TITLE
Громов Алексей. Вариант 2. Технология OMP. Умножение плотных матриц. Элементы типа double. Блочная схема, алгоритм Фокса.

### PR DESCRIPTION
**Pull request by DSFKnight**
`PreProcessingImpl()`
- Делит входные данные на две матрицы `𝐴` и `𝐵`
- Определяет размер матрицы 𝑛 и размер блока `block_size_`
- Проверяет, что матрица квадратная

`ValidationImpl():`
- Проверяет корректность входных данных: размеры должны быть чётными (две матрицы), сама матрица — квадратной

`RunImpl():`
- Делит матрицы на блоки размером `block_size_`
- Выполняет блочное умножение в несколько стадий с использованием OpenMP
- Использует три вычислительных цикла для обработки блоков: Внешний цикл по `i` параллелизуется через `#pragma omp parallel for default(none)`. Внутренние циклы по `bi`, `bj`, `bk` выполняются последовательно внутри каждого потока

`PostProcessingImpl():`
- Копирует результат из `output_` в выходной буфер

### CODE REVIEW

File: tasks/omp/gromov_a_fox_algorithm/func_tests/main.cpp [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+  gromov_a_fox_algorithm_omp::TestTaskOpenMP test_task_omp(task_data_omp);
+  ASSERT_TRUE(test_task_omp.Validation());
+  test_task_omp.PreProcessing();
+  test_task_omp.Run();
+  test_task_omp.PostProcessing();
+
+  for (size_t i = 0; i < out.size(); ++i) {
+    EXPECT_NEAR(out[i], expected[i], 1e-9);
+  }
+}
```
Comment (ofuuse) [inline tasks/omp/gromov_a_fox_algorithm/func_tests/main.cpp:178]: Add tests with negative values

**Review by ofuuse — CHANGES_REQUESTED** [review]

File: tasks/omp/gromov_a_fox_algorithm/func_tests/main.cpp [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+
+TEST(gromov_a_fox_algorithm_omp, test_50x50) {
```
Comment (myffii) [inline tasks/omp/gromov_a_fox_algorithm/func_tests/main.cpp:45]: Please, add random tests

File: tasks/omp/gromov_a_fox_algorithm/perf_tests/main.cpp [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+#include <cstdint>
+#include <memory>
+#include <vector>
+
+#include "core/perf/include/perf.hpp"
+#include "core/task/include/task.hpp"
+#include "omp/gromov_a_fox_algorithm/include/ops_omp.hpp"
+
+TEST(gromov_a_fox_algorithm_omp, test_pipeline_run) {
+  constexpr size_t kN = 400;
```
Comment (myffii) [inline tasks/omp/gromov_a_fox_algorithm/perf_tests/main.cpp:14]: increase values ​​in perf tests

**Review by myffii — CHANGES_REQUESTED** [review]

**Review by ofuuse — APPROVED** [review]

**Review by myffii — APPROVED** [review]

**Review by aobolensk — APPROVED** [review]

File: tasks/omp/gromov_a_fox_algorithm/perf_tests/main.cpp [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+#include <cstdint>
+#include <memory>
+#include <vector>
+
+#include "core/perf/include/perf.hpp"
+#include "core/task/include/task.hpp"
+#include "omp/gromov_a_fox_algorithm/include/ops_omp.hpp"
+
+TEST(gromov_a_fox_algorithm_omp, test_pipeline_run) {
+  constexpr size_t kN = 450;
```
[the remark this answers is not in the document]
Comment (DSFKnight) [reply tasks/omp/gromov_a_fox_algorithm/perf_tests/main.cpp]: My commit failed because of another student's work. Please check it. 
Снимок экрана (263)

**Review by allnes — APPROVED** [review]

File: tasks/omp/gromov_a_fox_algorithm/perf_tests/main.cpp (hunk 4 above)
[the remark this answers is not in the document]
Comment (allnes) [reply tasks/omp/gromov_a_fox_algorithm/perf_tests/main.cpp]: @DSFKnight https://github.com/learning-process/ppc-2025-threads/pull/388
